lib/lpc55-puf/src/lib.rs

```diff
@@ -315,6 +315,48 @@ impl<'a> Puf<'a> {
         }
     }
 
+    fn is_lock_bit_set(&self, idxblk: u32) -> bool {
+        (idxblk & 1 << 30) != 0
+    }
+
+    fn is_unlock_bit_set(&self, idxblk: u32) -> bool {
+        (idxblk & 1 << 31) != 0
+    }
+
+    fn is_locked(&self, idxblk: u32) -> bool {
+        if self.is_lock_bit_set(idxblk) && !self.is_unlock_bit_set(idxblk) {
```

Comment:
I suspect I'd find this easier to read/understand as `match (self.is_lock_bit_set(idxblk), self.is_unlock_bit_set(idxblk))`

Reply:
that is much easier to read